air/tests/test_module/instructions/ap.rs

```diff
@@ -185,3 +187,32 @@ fn ap_with_dst_stream() {
     assert_eq!(actual_trace, expected_state);
     assert!(result.next_peer_pks.is_empty());
 }
+
+#[test]
+fn par_ap_behaviour() {
+    let vm_1_peer_id = "vm_1_peer_id";
+    let vm_2_peer_id = "vm_2_peer_id";
+    let vm_3_peer_id = "vm_3_peer_id";
+    let mut vm_1 = create_avm(echo_call_service(), vm_1_peer_id);
+
+    let script = format!(
+        r#"
+        (par
+            (call "{2}" ("peer" "timeout") [] join_it)
+            (seq
+                (par
+                    (call "{0}" ("peer" "timeout") [join_it] $result)
+                    (ap "fast_result" $result)
+                )
+                (call "{1}" ("op" "return") [$result.$[0]])
+            )
+        )
```

Comment:
- Since `vm_1` is the one who initiates the script, `op return` should be executed there. 
- `peer timeout` should be executed on vm_2 for this test to keep original meaning
-  There's no vm_3 in the original test, but I think that doesn't matter much

```suggestion
        (par
            (call "{2}" ("peer" "timeout") [] join_it)
            (seq
                (par
                    (call "{1}" ("peer" "timeout") [join_it] $result)
                    (ap "fast_result" $result)
                )
                (call "{0}" ("op" "return") [$result.$[0]])
            )
        )
```